daphne/src/roles.rs

```diff
@@ -582,14 +580,33 @@ pub trait DapHelper<S>: DapAggregator<S> {
                     ));
                 }
 
+                // Remove reports that are rejected early.
+                let early_rejects = early_rejects_future.await?;
                 let agg_resp = match transition {
-                    DapHelperTransition::Continue(state, agg_resp) => {
+                    DapHelperTransition::Continue(mut state, mut agg_resp) => {
+                        let mut i = 0;
+                        while i < state.seq.len() {
+                            if let Some(failure) = early_rejects.get(&agg_resp.transitions[i].nonce)
+                            {
+                                // Mark reports that were rejected early as TransitionVar::Failed.
+                                agg_resp.transitions[i].var = TransitionVar::Failed(*failure);
+
+                                // Remove VdafState of reports that were rejected early.
+                                if state.seq[i].1 == agg_resp.transitions[i].nonce {
+                                    let _val = state.seq.remove(i);
+                                }
+                            } else {
+                                i += 1;
+                            }
+                        }
+
                         self.put_helper_state(
                             &agg_init_req.task_id,
                             &agg_init_req.agg_job_id,
                             &state,
                         )
                         .await?;
+
```

Comment:
nit: Remove spurious newline. (It's good practice to keep changes out of PRs that are not necessary.)

Reply:
Sorry :(